src/time-slot-bar/time-slot-bar.tsx

```diff
@@ -0,0 +1,307 @@
+import dayjs from "dayjs";
+import customParseFormat from "dayjs/plugin/customParseFormat";
+import { useEffect, useRef, useState } from "react";
+import {
+    ArrowButton,
+    ArrowIconLeft,
+    ArrowIconRight,
+    Border,
+    CELL_WIDTH,
+    CellText,
+    Container,
+    TimeLabel,
+    TimeMarker,
+    TimeMarkerWrapper,
+    TimeSlot,
+    TimeSlotBarContainer,
+    TimeSlotWrapper,
+} from "./time-slot-bar.styles";
+import { TimeSlotBarProps } from "./types";
+// Load plugins
+dayjs.extend(customParseFormat);
+
+export const TimeSlotBar = ({
+    testId,
+    startTime,
+    endTime,
+    slots,
+    onClickSlot,
+    defaultTimeSlot,
+}: TimeSlotBarProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const barRef = useRef<HTMLDivElement>(null);
+    const [scrollPosition, setScrollPosition] = useState<number>(0);
+    const [clientWidth, setClientWidth] = useState<number>(0);
+
+    const CELL_DURATION = 30; // In minutes
+    const SCROLL_INCREMENT = CELL_WIDTH * 2.5; // In px. Each scroll increment corresponds to 75mins
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEffect(() => {
+        const handleScroll = () => {
+            if (barRef.current) {
+                setScrollPosition(barRef.current.scrollLeft);
+            }
+        };
+
+        const container = barRef.current;
+        if (container) {
+            window.addEventListener("resize", handleResize);
```

Comment:
this doesnt seem to be working properly. Consider using the `useResizeDetector` by `react-resize-detector`. It's already installed in this repo